### PR TITLE
001 create login screen

### PR DESCRIPTION
**Pull request by dtngoc99**
## Change Description 👋

Create login screen

## Demo 📺

**[Loom Video Demo](https://www.loom.com/share/0ff7d947c0a74018a99ec9bd5b2b493d)**

Bring clarity to the code with visual aids:

-   **Screenshots:** `cmd` + `shift` + `4` _(MacOS)_.
-   **Gifs:** [GIPHY Capture](https://giphy.com/apps/giphycapture) _(free/MacOS)_.

### CODE REVIEW

**Comment by dtngoc99** [comment]
Demo: https://www.loom.com/share/97f529b4e7614d2db681f81c424988b1

File: src/main.js [hunk 1]
```diff
@@ -4,12 +4,31 @@ import "./registerServiceWorker";
...
   router,
   store,
+  created() {
+    //const userString = localStorage.getItem("user");
+    //if (userString) {
+    // const userData = JSON.parse(userString);
+    // this.$store.commit("SET_USER_DATA", userData);
+    //} else {
+    this.$router.push({ path: "/auth/signin" });
+    //}
```
Comment (luantc13) [inline src/main.js:21]: I think we have a wrong implementation here.
Don't like the course example, we just need to save the token to the local storage, `token` and `eYj...`.
and we need to check the existence of the token here to navigate to the right route, `/auth/signin` or `/companies`.

File: src/router/index.js [hunk 2]
```diff
@@ -1,23 +1,23 @@
...
+    component: () => import("../views/AuthPage.vue"),
+    children: [
+      {
+        path: "signin",
+        name: "Signin",
+        component: () => import("@/signin/views/Signin.vue"),
+      },
+      {
+        path: "singup",
+        component: () => import("../views/About.vue"),
```
Comment (luantc13) [inline src/router/index.js:18]: Two ways, we should use only one way for importing here.
Btw, we will need to update the SignUp component soon.

File: src/signin/store/signin.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+export const HTTP = axios.create({
+  baseURL: `http://intern-2021.southeastasia.cloudapp.azure.com:4000/api/v1`,
+});
+
+export const namespaced = true;
+
+export const mutations = {
+  SET_USER_DATA(state, userData) {
+    state.user = userData;
+    localStorage.setItem("user", JSON.stringify(userData));
```
Comment (luantc13) [inline src/signin/store/signin.js:15]: Just need to save the token to the local storage, no need to save the whole response.

File: src/views/AuthPage.vue [hunk 4]
```diff
@@ -0,0 +1,38 @@
+<template>
+  <div class="bg-main fill-height-flex">
+    <v-layout column class="fill-height">
+      <HeaderLayout class="header" />
+      <router-view class="content" />
+      <FooterLayout class="footer" />
```
Comment (luantc13) [inline src/views/AuthPage.vue:6]: We should rename `HeaderLayout` and `FooterLayout` to `AuthHeader` and `AuthFooter`.
The reason is that we can have many headers and footers in an application, HeaderLayout and FooterLayout is too general.

File: src/views/Home.vue [hunk 5]
```diff
@@ -1,5 +1,7 @@
 <template>
-  <div></div>
+  <div>
+    <h1>Ngọc Tỷ siêu dễ thương nhất hệ mặt trời</h1>
```
Comment (luantc13) [inline src/views/Home.vue:3]: Nhất vũ trụ luôn :wink:

**Review by luantc13 — COMMENTED** [review]
Greate work!
Having some comments, happy to merge when they are resolved.

**Comment by dtngoc99** [comment]
Update demo: https://www.loom.com/share/795b62005b3943d0aba042e01c4536bf

**Review by luantc13 — APPROVED** [review]
Great work!

**Pull request merged**